### PR TITLE
fix: Don't propagate sampled traces

### PR DESCRIPTION
**Pull request by loewenheim**
Fixes https://github.com/getsentry/team-ingest/issues/587.

### CODE REVIEW

File: relay-log/src/setup.rs [hunk 1]
```diff
@@ -294,14 +295,25 @@ pub fn init(config: &LogConfig, sentry: &SentryConfig) {
...
 
     if let Some(dsn) = sentry.enabled_dsn() {
+        let traces_sample_rate = config.traces_sample_rate;
+        // We're explicitly setting a `traces_sampler` here to circumvent trace
+        // propagation. A trace sampler that always just returns the constant
+        // `traces_sample_rate` is equivalent to using the `traces_sample_rate`
+        // directly, except it doesn't take into account whether the context
+        // was previously sampled. We don't want to take that into account because
+        // SDKs send headers with their envelopes that erroneously cause us to
+        // sample transactions.
```
Comment (loewenheim) [inline relay-log/src/setup.rs:305]: Please let me know if this comment doesn't explain the problem correctly or clearly enough.

File: relay-log/src/setup.rs [hunk 2]
```diff
@@ -294,14 +295,25 @@ pub fn init(config: &LogConfig, sentry: &SentryConfig) {
...
+            Some(Arc::new(move |_: &TransactionContext| traces_sample_rate) as Arc<TracesSampler>);
         let mut options = sentry::ClientOptions {
             dsn: Some(dsn).cloned(),
             in_app_include: vec!["relay"],
             release: Some(RELEASE.into()),
             attach_stacktrace: config.enable_backtraces,
             environment: sentry.environment.clone(),
             server_name: sentry.server_name.clone(),
-            traces_sample_rate: config.traces_sample_rate,
+            traces_sample_rate,
```
Comment (loewenheim) [inline relay-log/src/setup.rs]: Should we even still set the `traces_sample_rate` here? If a `traces_sampler` is set, it's effectively dead.

File: relay-log/src/setup.rs [hunk 3]
```diff
@@ -317,6 +329,14 @@ pub fn init(config: &LogConfig, sentry: &SentryConfig) {
             }));
         }
 
+        crate::info!(
+            release = RELEASE,
+            server_name = ?sentry.server_name,
+            environment = ?sentry.environment,
```
Comment (loewenheim) [inline relay-log/src/setup.rs]: I would love these fields to be added conditionally (only if they're `Some`), but `tracing` doesn't seem to support that.

**Review by Dav1dde — APPROVED** [review]

File: relay-log/src/setup.rs (hunk 2 above)
Comment (Dav1dde) [reply relay-log/src/setup.rs]: I am quite indifferent here, can also remove it to make it explicit the traces sampler is the only one making decisions.

File: relay-log/src/setup.rs (hunk 3 above)
Comment (Dav1dde) [reply relay-log/src/setup.rs]: That should be the default behaviour: https://docs.rs/tracing-core/0.1.32/src/tracing_core/field.rs.html#749

Comment (Dav1dde) [reply relay-log/src/setup.rs]: Ah maybe doesn't work here because you convert to `Debug` representation with the `?`, if you drop it, it should work.

Comment (loewenheim) [reply relay-log/src/setup.rs]: I misinterpreted the error message. I though `Option` didn't implement `Value`, but actually `Cow<str>` is the problem.

File: relay-log/src/setup.rs (hunk 2 above)
Comment (loewenheim) [reply relay-log/src/setup.rs]: Ok, I removed it.

**Pull request merged**